src/adapterService/edgeAdapterService.ts

```diff
@@ -1,442 +1,426 @@
-                        sockets[i].close();
-                    }
-                }
-                this._edgeToWSMap.delete(instanceId);
-                this._guidToIdMap.delete(guid.toLocaleUpperCase());
-                this._guidToIdMap.delete(id);
-                this._idToNetWorkProxyMap.delete(id);
-                this._idToNetWorkProxyMap.delete(networkInstanceId);
-                this._idToEdgeMap.delete(id);
-            }
-            return edgeResult && networkProxyResult;
-        }
-    }
-}
+﻿///<reference path='../../lib/Addon.node.d.ts' />
+//
+// Copyright (C) Microsoft. All rights reserved.
+//
+
+import { IChromeInstance } from '../../lib/EdgeAdapterInterfaces';
+import * as edgeAdapter from '../../lib/Addon.node';
+// const binding_path = binary.find(path.resolve(path.join(__dirname,'../../../package.json')));
+// var edgeAdapter = require(binding_path);
+
+import * as http from 'http';
+import * as ws from 'ws';
+import * as fs from 'fs';
+import * as crypto from 'crypto';
+import { Server as WebSocketServer } from 'ws';
+
+declare var __dirname;
+
+export module EdgeAdapter {
+    export class Service {
+        private _httpServer: http.Server;
+        private _webSocketServer: WebSocketServer;
+        private _serverPort: number;
+        private _chromeToolsPort: number;
+        private _guidToIdMap: Map<string, string> = new Map<string, string>();
+        private _idToEdgeMap: Map<string, edgeAdapter.EdgeInstanceId> = new Map<string, edgeAdapter.EdgeInstanceId>();
+        private _idToNetWorkProxyMap: Map<string, edgeAdapter.NetworkProxyInstanceId> = new Map<string, edgeAdapter.NetworkProxyInstanceId>();
+        private _edgeToWSMap: Map<edgeAdapter.EdgeInstanceId, ws[]> = new Map<edgeAdapter.EdgeInstanceId, ws[]>();
+        private _diagLogging: boolean = false;
+        private _newtworkProxyImplementedRequests: Array<string> = new Array<string>('Network.enable', 'Network.disable', 'Network.getResponseBody');
+
+        constructor(diagLogging: boolean) {
+            this._diagLogging = diagLogging;
+        }
+
+        public run(serverPort: number, chromeToolsPort: number, url: string): void {
+            this._serverPort = serverPort;
+            this._chromeToolsPort = chromeToolsPort;
+
+            edgeAdapter.initialize(__dirname, (id, msg) => this.onEdgeMessage(id, msg), (msg) => this.onLogMessage(msg));
+            edgeAdapter.setSecurityACLs(__dirname + "\\..\\..\\lib\\");
+
+            if (chromeToolsPort > 0) {
+                edgeAdapter.serveChromeDevTools(chromeToolsPort);
+            }
+
+            if (url) {
+                edgeAdapter.openEdge(url);
+            }
+
+            this._httpServer = http.createServer((req, res) => this.onServerRequest(req, res));
+            this._webSocketServer = new WebSocketServer({ server: this._httpServer });
+            this._webSocketServer.on('connection', (client) => this.onWSSConnection(client));

```

Comment:
you need to add a second parameter to the event handler. Something like this:
`this._webSocketServer.on('connection', (client, message) => this.onWSSConnection(client, message));`

Reply:
fixed and `protocol.json` updated. I'll need to manually update the release with the new files though